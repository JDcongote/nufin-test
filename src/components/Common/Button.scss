@import '../../styles/variables/index';
@import '../../styles/tools/mixins';

.button {
  @include button-unstyled;
  z-index: var(--z-index--level-1);
  border-radius: var(--border--radius--small);

  &-primary {
    padding: 0.5em;
    background-color: var(--color--button-primary);
    font-weight: bold;
    color: var(--color--secondary);
  }
  &.bottom {
    position: fixed;
    bottom: 0;
    margin: 1em;
    &-right {
      right: 0;
    }
    &-left {
      left: 0;
    }
  }
}
