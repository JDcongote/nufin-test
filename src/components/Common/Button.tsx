import React from 'react';
import './Button.scss';

type ButtonProps = {
  onClick: (data: any) => void;
  text?: string;
  context: any;
};

const Button = ({ text, onClick, context }: ButtonProps) => {
  function handleCLick(event: React.MouseEvent) {
    onClick(context);
  }
  return (
    <button className="button button-primary" onClick={handleCLick}>
      {text}
    </button>
  );
};

export default Button;
