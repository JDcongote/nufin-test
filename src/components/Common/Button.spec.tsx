import Button from './Button';
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Button', () => {
  it('Basic Button Test', () => {
    const onClick = () => {};
    const text = 'TEST_BUTTON';
    const context = { title: 'TEST' };
    const wrapper = mount(
      <Button context={context} onClick={onClick} text={text} />
    );
    const render = renderer.create(
      <Button context={context} onClick={onClick} text={text} />
    );

    const nodes = wrapper.find('.button');
    expect(nodes.text()).toBe(text);

    expect(render.toJSON()).toMatchSnapshot();
  });

  it('onCLick Test', () => {
    const onClick = jest.fn();
    const text = 'TEST_BUTTON';
    const context = { title: 'TEST' };
    const wrapper = mount(
      <Button context={context} onClick={onClick} text={text} />
    );

    const button = wrapper.find('.button');
    button.simulate('click');
    expect(onClick).toBeCalledWith(context);
  });
});
