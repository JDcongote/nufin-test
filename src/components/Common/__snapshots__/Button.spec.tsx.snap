// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Basic Button Test 1`] = `
<button
  className="button button-primary"
  onClick={[Function]}
>
  TEST_BUTTON
</button>
`;
